src/statefulmodelinstance.hpp

```diff
@@ -0,0 +1,61 @@
+//*****************************************************************************
+// Copyright 2021 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+#pragma once
+
+#include <memory>
+#include <string>
+
+#include "modelinstance.hpp"
+
+namespace ovms {
+
+const uint32_t NO_CONTROL_INPUT = 0;
+const uint32_t SEQUENCE_START = 1;
+const uint32_t SEQUENCE_END = 2;
+
+/**
+     * @brief This class contains all the information about inference engine model
+     */
+class StatefulModelInstance : public ModelInstance {
+public:
+    /**
+         * @brief A default constructor
+         */
+    StatefulModelInstance(const std::string& name, model_version_t version) :
```

Comment:
Is it required? Won't base class constructor be called implicitly?

Reply:
I remember it was required for some tests or model factory.